### ISSUE TITLE
vcpkg install exits without downloading package

### ISSUE THREAD

**Issue by aagarwl3** [microsoft/vcpkg#10830]
**Describe the bug**
After successfully executing bootstrap_vcpkg.bat; vcpkg integrate install; vcpkg integrate powershell, I moved to installing zlib using "vcpkg install zlib:x64-windows" but it fails to download and exits without any explicit error message.  The on-screen outputs and abi info file that is generated is attached.  

Following the debugging of other issues I executed "vcpkg env --debug" which also fails to launch/set up the VS 2017 developer environment.  

I have separately tried to execute the last command (cmd /c ""C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Auxiliary\Build\vcvarsall.bat" -vcvars_ver=14.1 amd64   2>&1 <NUL" & echo cdARN4xjKueKScMy9C6H& set) which executes fine.

Please advise how to fix this.

**Environment**
- OS: Windows 10 Enterprise (Build 17134)
- Compiler: Microsoft Visual Studio Professional 2017 (Version 15.9.21)

**Failure logs** 
vcpkg.install.notes -- on-screen outputs of the set up process
x64-windows.vcpkg_abi_info -- abi info file recorded by attempting to install in debug mode
[x64-windows.vcpkg_abi_info.txt](https://github.com/microsoft/vcpkg/files/4477822/x64-windows.vcpkg_abi_info.txt)
[vcpkg.install.log](https://github.com/microsoft/vcpkg/files/4477826/vcpkg.install.log)

**Additional context**
I have the same environment on another laptop and it works fine (although the vcpkg is an older build) so I do not believe it has anything to do with my internet and/or firewall.  The same steps have been executed successfully by others on the team without any issues.

**Comment by NancyLi1013** [comment]
Hi @aagarwl3 
Thanks for posting this issue.
I cannot reproduce this on my local with Visual Studio 2017 and Visual Studio 2019.

**Comment by aagarwl3** [comment]
Hi @NancyLi1013 
Thanks for following up.  Reproduction has indeed been a challenge.  As I noted, others in my team have not faced similar issues and so it has been difficult to understand where the gap is.  Any other information that you'd like to see perhaps?  

I am not following what the system.process.cpp is expecting on line 487 and causing execution to stop.  Thanks!

**Comment by NancyLi1013** [comment]
Could you please help confirm if you updated to the latest commit?
Does this issue exist on the old version or the latest version?

**Comment by aagarwl3** [comment]
Hi @NancyLi1013 
I am using the latest commit (commit 941d5464544eb5812b2cbd6e687c3ebc78ed2624).  The issue does not happen on another laptop on which I have an older version of vcpkg (from 2019).

**Comment by NancyLi1013** [comment]
Thanks for your update.
Since vcpkg updates some changes recently.
Could you please update to the latest commit [09af1e9](https://github.com/microsoft/vcpkg/commit/09af1e9b55fdb79ef5aa04de3f1710759b2de990) to try it again?

**Comment by aagarwl3** [comment]
Hi @NancyLi1013 
Thanks for the info.  I updated to the latest commit but unfortunately no change in behavior.  

I forgot to include a behavior I am noticing at the end of some processes (integrate install; integrate powershell; install zlib):  there seems to be a command prompt being launched in the background (which exits automatically) since I have an autorun set up to my command processor in the registry.  The doskeys associated in the autorun seem to be executed.  For clarity I have also tested by removing the doskeys without success.
